pkg/controller/configmap/config.go

```diff
@@ -310,24 +322,34 @@ func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap)
 			}
 		}
 	}
-	// Check whether different sections impact on NCP and nsx-node-agent
-	ncpNeedChange, agentNeedChange = false, false
+	// Check whether different sections impact on ncp, nsx-node-agent and nsx-ncp-bootstrap
+	needChange := podNeedChange{}
 	for _, sec := range diffSecs {
-		if !ncpNeedChange && inSlice(sec, operatortypes.NcpSections) {
-			ncpNeedChange = true
+		if !needChange.ncp && inSlice(sec, operatortypes.NcpSections) {
+			needChange.ncp = true
+		}
+		if !needChange.agent && inSlice(sec, operatortypes.AgentSections) {
+			needChange.agent = true
 		}
-		if !agentNeedChange && inSlice(sec, operatortypes.AgentSections) {
-			agentNeedChange = true
+		bootstrapOpts, found := operatortypes.BootstrapOptions[sec]
+		if !needChange.bootstrap && found {
+			for _, opt := range bootstrapOpts {
+				if currCfg.Section(sec).Key(opt).Value() != prevCfg.Section(sec).Key(opt).Value() {
```

Comment:
could there be an error here if the `Key` is not found in prev or curr cfg?

Reply:
If Key is not found, it will return a zero-value. 
Refer to https://godoc.org/gopkg.in/go-ini/ini.v1#Section.Key